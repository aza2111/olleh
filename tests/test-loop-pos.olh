//tests loops with multi-statement bodies

int x = 3;
while (x > 0):
        print x;
        x = x - 1;
        print x;
end;

for (5):
        print "five times this statement";
        print "and this one";
        print "and the other";
end;

foreach z ['a', 'b', 'c', 'd']:
        if (z > 'b'):
                print "z is greater than b";
        end;
        print z;
end;
