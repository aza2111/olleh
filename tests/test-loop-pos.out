3
2
2
1
1
0
five times this statement
and this one
and the other
five times this statement
and this one
and the other
five times this statement
and this one
and the other
five times this statement
and this one
and the other
five times this statement
and this one
and the other
'a'
'b'
z is greater than b
'c'
z is greater than b
'd'
